packages/@react-spectrum/table/test/TableBlah.test.tsx

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {act, render, screen} from '@react-spectrum/test-utils';
+import {Cell, Column, Row, TableBody, TableHeader, TableView} from '../';
+import {Provider} from '@react-spectrum/provider';
+import React, {useState} from 'react';
+import {TableTester} from '@react-aria/test-utils';
+import {theme} from '@react-spectrum/theme-default';
+
+let manyItems = [];
+for (let i = 1; i <= 100; i++) {
+  manyItems.push({id: i, foo: 'Foo ' + i, bar: 'Bar ' + i, baz: 'Baz ' + i});
+}
+
+let columns = [
+  {name: 'Foo', key: 'foo'},
+  {name: 'Bar', key: 'bar'},
+  {name: 'Baz', key: 'baz'}
+];
+
+describe('Table ', function () {
+  let offsetWidth, offsetHeight;
+  let onSelectionChange = jest.fn();
+  let onSortChange = jest.fn();
+
+  let TableExample = (props) => {
+    let [sort, setSort] = useState({});
+    let setSortDescriptor = (sort) => {
+      setSort(sort);
+      onSortChange(sort);
+    };
+
+    return (
+      <Provider theme={theme}>
+        <TableView aria-label="Table" selectionMode="multiple" data-testid="test" sortDescriptor={sort} onSortChange={setSortDescriptor} onSelectionChange={onSelectionChange} {...props}>
+          <TableHeader columns={columns}>
+            {column => <Column allowsSorting allowsResizing={props.allowsResizing}>{column.name}</Column>}
+          </TableHeader>
+          <TableBody items={manyItems}>
+            {item =>
+              (<Row key={item.foo}>
+                {key => <Cell>{item[key]}</Cell>}
+              </Row>)
+            }
+          </TableBody>
+        </TableView>
+      </Provider>
+    );
+  };
+
+  beforeAll(function () {
```

Comment:
this is probably on your radar, but having these mocks and such callable with 1 line of code would be nice too!

Reply:
Yep, I've been mulling over whether or not these mocks are too broad. Ideally we would limit what it actually gets applied on, similar to https://github.com/adobe/react-spectrum/blob/1c32cd5559048ace7b185f4c594d503aea0e6ae7/packages/%40react-aria/overlays/test/useOverlayPosition.test.tsx#L244-L252, but I'm worried that users may need to have finer control over the mocks.

Perhaps we can provide a basic util that does the simple mock and offer a way to override/customize

Reply:
Another interesting idea that @devongovett brought up was making our virtualized components render as non-virtualized components when rendered in a test environment specifically. That means we won't have to mock any of this stuff, thus avoid possible mock collisions and assumptions. RSP would still write tests with the virtualized components still being virtualized just to make sure things work as expected